### PR TITLE
HPOS - fatal error when trying to create a manual order in WC

### PR DESCRIPTION
**Pull request by mntzrr**
Fixed a critical order when the merchant creates a new order while HPOS is enabled.

Related task: https://app.clickup.com/t/865d18nj6

### CODE REVIEW

**Comment by mntzrr** [comment]
Task linked: [CU-865d18nj6 HPOS - fatal error when trying to create a manual order in WC](https://app.clickup.com/t/865d18nj6)

File: classes/class-wc-klarna-meta-box.php [hunk 1]
```diff
@@ -46,20 +43,11 @@ public function __construct() {
...
-
-		// CustomOrdersTableController was introduced in WC 6.4.
-		if ( class_exists( CustomOrdersTableController::class ) ) {
-			$hpos_enabled = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled();
-		}
-
-		$screen   = $hpos_enabled ? wc_get_page_screen_id( 'shop-order' ) : $post_type;
-		$order_id = $hpos_enabled ? filter_input( INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT ) : get_the_ID();
-
 		if ( in_array( $post_type, array( 'woocommerce_page_wc-orders', 'shop_order' ) ) ) {
```
Comment (MichaelBengtsson) [inline classes/class-wc-klarna-meta-box.php:46]: Add true as the third argument to fix the PHPCS issue to the in_array check

File: includes/klarna-order-management-functions.php [hunk 2]
```diff
@@ -9,6 +9,38 @@
...
 }
 
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+
+/**
+ * Equivalent to WP's get_the_ID() with HPOS support.
+ *
+ * @return int the order ID or false.
+ */
+//phpcs:ignore
```
Comment (MichaelBengtsson) [inline includes/klarna-order-management-functions.php:19]: What is this ignore for? Can you add a more specific rule for it to ignore with a description of why?

**Review by MichaelBengtsson — CHANGES_REQUESTED** [review]
Looks good, but please look over the PHPCS issues and solve them if possible

**Review by MichaelBengtsson — APPROVED** [review]

**Pull request merged**